lib/github_changelog_generator/parser.rb

```diff
@@ -1,165 +1,181 @@
-          options[:max_issues] = max
+        opts.on('--max-issues [NUMBER]', Integer, 'Max number of issues to fetch from GitHub. Default is unlimited') do |max|
+          @@options[:max_issues] = max
         end
-        opts.on("--release-url [URL]", "The URL to point to for release links, in printf format (with the tag as variable).") do |url|
-          options[:release_url] = url
+        opts.on('--release-url [URL]', 'The URL to point to for release links, in printf format (with the tag as variable).') do |url|
+          @@options[:release_url] = url
         end
-        opts.on("--github-site [URL]", "The Enterprise Github site on which your project is hosted.") do |last|
-          options[:github_site] = last
+        opts.on('--github-site [URL]', 'The Enterprise Github site on which your project is hosted.') do |last|
+          @@options[:github_site] = last
         end
-        opts.on("--github-api [URL]", "The enterprise endpoint to use for your Github API.") do |last|
-          options[:github_endpoint] = last
+        opts.on('--github-api [URL]', 'The enterprise endpoint to use for your Github API.') do |last|
+          @@options[:github_endpoint] = last
         end
-        opts.on("--simple-list", "Create simple list from issues and pull requests. Default is false.") do |v|
-          options[:simple_list] = v
+        opts.on('--simple-list', 'Create simple list from issues and pull requests. Default is false.') do |v|
+          @@options[:simple_list] = v
         end
-        opts.on("--future-release [RELEASE-VERSION]", "Put the unreleased changes in the specified release number.") do |future_release|
-          options[:future_release] = future_release
+        opts.on('--future-release [RELEASE-VERSION]', 'Put the unreleased changes in the specified release number.') do |future_release|
+          @@options[:future_release] = future_release
         end
-        opts.on("--release-branch [RELEASE-BRANCH]", "Limit pull requests to the release branch, such as master or release") do |release_branch|
-          options[:release_branch] = release_branch
+        opts.on('--release-branch [RELEASE-BRANCH]', 'Limit pull requests to the release branch, such as master or release') do |release_branch|
+          @@options[:release_branch] = release_branch
         end
-        opts.on("--[no-]verbose", "Run verbosely. Default is true") do |v|
-          options[:verbose] = v
+        opts.on('--[no-]verbose', 'Run verbosely. Default is true') do |v|
+          @@options[:verbose] = v
         end
-        opts.on("-v", "--version", "Print version number") do |_v|
+        opts.on('-v', '--version', 'Print version number') do |_v|
           puts "Version: #{GitHubChangelogGenerator::VERSION}"
           exit
         end
-        opts.on("-h", "--help", "Displays Help") do
+        opts.on('-h', '--help', 'Displays Help') do
```

Comment:
Alright, pushed a change. To each their own. In any case, this push request wasn't meant to be merged, it was more a discussion starter. For one, I think most tests would fail, but I haven't yet figured out how to run the tests.